funfact/lang/_terminal.py

```diff
@@ -153,7 +154,8 @@ class TensorSymbol(Symbol):
         _anon_registry = {}
         _anon_registry_lock = multiprocessing.Lock()
 
-    def __init__(self, *size, symbol=None, initializer=None, optimizable=True):
+    def __init__(self, *size, symbol=None, initializer=None, optimizable=True,
+                 prefer=NoCondition()):
```

Comment:
I'd prefer `prefer` (prefer x2 :grin:) to have a default value of `None`, which gets translated into `NoCondition` later on . This helps to simplify the function signature and makes the documentation more readable, since users will not need to figure out what `NoCondition` means.

Reply:
In the current implementation, there is actually a subtle difference between tensors that have `prefer=None` and the ones that have `prefer=NoCondition()`. The former is meant for non optimizable tensors for which the preference is not considered in the optimization, the latter is for optimizable tensors without a specified preference. Does that distinction make sense to you? I wanted to make this more explicit by changing line 22 of the `PenaltyEvaluator` to `return []`.

Reply:
I see. Since constraints is not meaningful when a tensor is non-optimizable, could we check for the `optimizable` flag when evaluating the penalties, and simply return `tensor(0, dtype=...)` upon True? In the docstring we can just say that the `prefer` keyword is ignored if the optimizable flag is False. Similarly, if `prefer` is None, then we just return a 0 tensor without further actions.

Reply:
ok, I have resolved this and made some changes in the `tensor` method in `_tsrex.py`. See comment there.